scripts/easyphoto_infer.py

```diff
@@ -199,144 +196,280 @@ def inpaint_only(
-        else:
-            generate_image = inpaint_only(output_image, input_mask, input_prompt, diffusion_steps=second_diffusion_steps, denoising_strength=second_denoising_strength, hr_scale=default_hr_scale)
-
-        # use original template face area to shift generated face color at last
-        if color_shift_last:
-            rescale_retinaface_box = [int(i * default_hr_scale) for i in retinaface_box]
-            output_image_face_area = np.array(copy.deepcopy(generate_image))[rescale_retinaface_box[1]:rescale_retinaface_box[3], rescale_retinaface_box[0]:rescale_retinaface_box[2],:] 
-            output_image_face_area = color_transfer(output_image_face_area,template_image_original_face_area)
-            generate_image = np.array(generate_image)
-            generate_image[rescale_retinaface_box[1]:rescale_retinaface_box[3], rescale_retinaface_box[0]:rescale_retinaface_box[2],:] = output_image_face_area
-            generate_image = Image.fromarray(generate_image)
-
-
-        # If it is a large template for cutting, paste the reconstructed image back
-        if crop_face_preprocess:
-            origin_image    = np.array(copy.deepcopy(template_image))
-            x1,y1,x2,y2     = crop_safe_box
-            generate_image  = generate_image.resize([x2-x1, y2-y1], Image.Resampling.LANCZOS)
-            origin_image[y1:y2,x1:x2] = np.array(generate_image) 
-            origin_image    = Image.fromarray(np.uint8(origin_image))
-        else:
-            origin_image    = generate_image
-        
+            template_image = Image.fromarray(template_image).convert("RGB")
+
+        template_face_safe_boxes, _, _ = call_face_crop(retinaface_detection, template_image, 1.10, "crop")
+        if len(template_face_safe_boxes) == 0:
+            return "Please upload a template with face.", []
+
+        if min(len(template_face_safe_boxes), len(user_ids)) > 1:
+            output_image = np.array(copy.deepcopy(template_image))
+            output_mask  = np.ones_like(output_image)
+
+            # get mask in final diffusion for multi people
+            for index in range(len(template_face_safe_boxes)):
+                retinaface_box = template_face_safe_boxes[index]
+                output_mask[retinaface_box[1]:retinaface_box[3], retinaface_box[0]:retinaface_box[2]] = 255
+            output_mask  = Image.fromarray(np.uint8(cv2.dilate(np.array(output_mask), np.ones((64, 64), np.uint8), iterations=1) - cv2.erode(np.array(output_mask), np.ones((32, 32), np.uint8), iterations=1)))
+
+        for index in range(min(len(template_face_safe_boxes), len(user_ids))):
+            loop_template_image = copy.deepcopy(template_image)
+            
+            # mask other people in this term
+            if min(len(template_face_safe_boxes), len(user_ids)) > 1:
```

Comment:
TODO: this code may cause error when face and detection box are overlapped!
@bubbliiiing  you should check this !